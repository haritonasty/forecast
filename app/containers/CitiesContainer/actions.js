/*
 *
 * CitiesContainer actions
 *
 */

import {
  ADD_NEW_CITY,
  CHANGE_ADDRESS,
  CHANGE_CURRENT_CITY,
  GET_CITY_INFO,
} from './constants';

export function getCityInfo(cities, city) {
  return {
    type: GET_CITY_INFO,
    cities,
    city,
  };
}

export function addNewCity(city) {
  return {
    type: ADD_NEW_CITY,
    city,
  };
}

export function changeAddress(address) {
  return {
    type: CHANGE_ADDRESS,
    address,
  };
}

export function changeCurrentCity(city) {
  return {
    type: CHANGE_CURRENT_CITY,
    city,
  };
}
