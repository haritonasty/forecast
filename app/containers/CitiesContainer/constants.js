/*
 *
 * CitiesContainer constants
 *
 */

export const ADD_NEW_CITY = 'app/CitiesContainer/ADD_NEW_CITY';
export const GET_CITY_INFO = 'app/CitiesContainer/GET_CITY_INFO';
export const CHANGE_ADDRESS = 'app/CitiesContainer/CHANGE_ADDRESS';
export const CHANGE_CURRENT_CITY = 'app/CitiesContainer/CHANGE_CURRENT_CITY';
