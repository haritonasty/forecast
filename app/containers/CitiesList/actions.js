/*
 *
 * CitiesList actions
 *
 */

import {
  ADD_NEW_CITY,
  CHANGE_CURRENT_CITY,
  INIT_CITIES,
  INIT_CITIES_LIST,
  REMOVE_CITY,
} from './constants';

export function addNewCity(id, city) {
  return {
    type: ADD_NEW_CITY,
    id,
    city,
  };
}

export function changeCurrentCity(cityId) {
  return {
    type: CHANGE_CURRENT_CITY,
    cityId,
  };
}

export function removeCity(cityId) {
  return {
    type: REMOVE_CITY,
    cityId,
  };
}

export function initCities() {
  return {
    type: INIT_CITIES,
  };
}

export function initCitiesList(cities) {
  return {
    type: INIT_CITIES_LIST,
    cities,
  };
}
