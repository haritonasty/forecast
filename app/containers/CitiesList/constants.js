/*
 *
 * CitiesList constants
 *
 */

export const ADD_NEW_CITY = 'app/SearchCitiesContainer/ADD_NEW_CITY';
export const CHANGE_CURRENT_CITY = 'app/SearchCitiesContainer/CHANGE_CURR_CITY';
export const REMOVE_CITY = 'app/SearchCitiesContainer/REMOVE_CITY';
export const INIT_CITIES = 'app/SearchCitiesContainer/INIT_CITIES';
export const INIT_CITIES_LIST = 'app/SearchCitiesContainer/INIT_CITIES_LIST';
