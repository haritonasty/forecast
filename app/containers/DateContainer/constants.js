/*
 *
 * DateContainer constants
 *
 */

export const CHANGE_DATE = 'app/DateContainer/CHANGE_DATE';
