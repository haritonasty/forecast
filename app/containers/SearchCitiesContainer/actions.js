/*
 *
 * SearchCitiesContainer actions
 *
 */

import {
  CHANGE_ADDRESS,
  GET_CITY_INFO,
  SET_ERROR,
  WAS_ERROR,
} from './constants';

export function getCityInfo(city) {
  return {
    type: GET_CITY_INFO,
    city,
  };
}

export function changeAddress(address) {
  return {
    type: CHANGE_ADDRESS,
    address,
  };
}

export function setError(error) {
  return {
    type: SET_ERROR,
    error,
  };
}

export function wasError(error) {
  return {
    type: WAS_ERROR,
    error,
  };
}
