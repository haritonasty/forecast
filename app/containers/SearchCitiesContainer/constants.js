/*
 *
 * SearchCitiesContainer constants
 *
 */

const common = 'app/SearchCitiesContainer/';

export const GET_CITY_INFO = `${common}GET_CITY_INFO`;
export const CHANGE_ADDRESS = `${common}CHANGE_ADDRESS`;
export const SET_ERROR = `${common}SET_ERROR`;
export const WAS_ERROR = `${common}WAS_ERROR`;
