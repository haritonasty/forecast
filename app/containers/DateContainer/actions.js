/*
 *
 * DateContainer actions
 *
 */

import { CHANGE_DATE } from './constants';

export function changeDate(value) {
  return {
    type: CHANGE_DATE,
    value,
  };
}
