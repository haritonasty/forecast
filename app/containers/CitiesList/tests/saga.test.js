import { put, takeEvery } from 'redux-saga/effects';
import { changeCurrentCity, initCitiesList } from '../actions';
import { INIT_CITIES } from '../constants';
import CitiesSaga, { initCities } from '../saga';

describe('CitiesSaga Saga', () => {
  const citiesSaga = CitiesSaga();

  it('should start task to watch for INIT_CITIES action', () => {
    const takeEveryDescriptor = citiesSaga.next().value;
    expect(takeEveryDescriptor).toEqual(takeEvery(INIT_CITIES, initCities));
  });
});

describe('initCities Saga', () => {
  let initCitiesGenerator;

  beforeEach(() => {
    initCitiesGenerator = initCities();

    const selectDescriptor = initCitiesGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the initCitiesList, changeCurrentCity actions if it requests the data successfully', () => {
    const citiesList = {
      '1': { city: 'moscow', country: 'Russia', lan: 5, lng: 4, id: 1 },
    };
    const currentCity = '1';
    const response = { citiesList, currentCity };
    const putDescriptor = initCitiesGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(initCitiesList(citiesList)));
    const putDescriptor2 = initCitiesGenerator.next(response).value;
    expect(putDescriptor2).toEqual(put(changeCurrentCity(currentCity)));
  });
});
