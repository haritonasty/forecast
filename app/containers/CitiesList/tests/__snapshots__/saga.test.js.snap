// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initCities Saga should dispatch the initCitiesList, changeCurrentCity actions if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/api",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
